shopinvader_product_variant_selector/models/shopinvader_variant.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Akretion (http://www.akretion.com).
+# @author Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class ShopinvaderVariant(models.Model):
+    _inherit = "shopinvader.variant"
+
+    variant_selector = fields.Serialized(
+        compute="_compute_variant_selector", string="Shopinvader Selector"
+    )
+
+    # TODO we should do an integration with
+    # shopinvader_product_state to be able to show the
+    # stock information
+    # shopinvader_product_stock_variant_selector
+    # should inherit this method
+    def _prepare_selector_value(self, variant, value):
+        return {
+            "name": value.name,
+            "sku": variant.default_code or "",
+            "available": variant.active,
+            "selected": variant == self,
+        }
+
+    def _get_matching_variant(self, values):
+        """Return the first variant that matches the values."""
+        for variant in self.shopinvader_variant_ids:
+            if values <= variant.attribute_value_ids:
```

Comment:
@sebastienbeau  what's the meaning of `<=` operator?

Reply:
it's True if all records of "variant.attribute_value_ids" are include into values.